### PR TITLE
Support the share_with and copy_to parameters as extended_attributes.

### PR DESCRIPTION
**Pull request by kalymero**
The packer part is quite ugly since it's not possible to pass a list on the command line as `var`.
Accounts and regions must me passed one by one

### CODE REVIEW

**Comment by ajordens** [comment]
Needs a rebase @kalymero.

Can you give it a once over @duftler, I'm not a `rosco` user but in theory it seems sensible to me.

`ami_users` is a little less important (right?) given that Spinnaker will on-demand share to an account (_ie. it doesn't need to be shared at bake time_).

File: rosco-core/src/main/groovy/com/netflix/spinnaker/rosco/providers/CloudProviderBakeHandler.groovy [hunk 1]
```diff
@@ -168,14 +168,33 @@ abstract class CloudProviderBakeHandler {
...
+      parameterMap << bakeRequest.extended_attributes.subMap(
+              attributes.findAll{ !it.equals('share_with') && !it.equals('copy_to') })
     }
 
     def finalTemplateFileName = bakeRequest.template_file_name ?: templateFileName
 
     return packerCommandFactory.buildPackerCommand(baseCommand, parameterMap, "$configDir/$finalTemplateFileName")
   }
 
+  private void unrollParameters(String prefix, String rolledParameter, Map parameterMap) {
```
Comment (duftler) [inline rosco-core/src/main/groovy/com/netflix/spinnaker/rosco/providers/CloudProviderBakeHandler.groovy]: Can drop the semi-colons in this method.

File: rosco-core/src/main/groovy/com/netflix/spinnaker/rosco/providers/CloudProviderBakeHandler.groovy [hunk 2]
```diff
@@ -168,14 +168,33 @@ abstract class CloudProviderBakeHandler {
...
+    List<String> values = rolledParameter.tokenize(",")
+    int i=1;
```
Comment (duftler) [inline rosco-core/src/main/groovy/com/netflix/spinnaker/rosco/providers/CloudProviderBakeHandler.groovy]: Would be more idiomatic to use `eachWithIndex` (http://docs.groovy-lang.org/latest/html/groovy-jdk/java/lang/Object.html#eachWithIndex(groovy.lang.Closure) here.

File: rosco-core/src/test/groovy/com/netflix/spinnaker/rosco/providers/aws/AWSBakeHandlerSpec.groovy [hunk 3]
```diff
@@ -859,4 +859,108 @@ class AWSBakeHandlerSpec extends Specification {
       bakeKey1 == "bake:aws:centos:kato|nflx-djangobase-enhanced_0.1-h12.170cdbd_all|mongodb:us-east-1:hvm:enhancedNWDisabled"
       bakeKey2 == bakeKey1
   }
+
```
Comment (duftler) [inline rosco-core/src/test/groovy/com/netflix/spinnaker/rosco/providers/aws/AWSBakeHandlerSpec.groovy]: Indentation in these new test methods doesn't match the neighbors.

**Comment by duftler** [comment]
LGTM once all comments are addressed.

File: rosco-core/src/main/groovy/com/netflix/spinnaker/rosco/providers/CloudProviderBakeHandler.groovy (hunk 2 above)
Comment (kalymero) [inline rosco-core/src/main/groovy/com/netflix/spinnaker/rosco/providers/CloudProviderBakeHandler.groovy]: ohhh yes

**Comment by kalymero** [comment]
it should be fine now

**Comment by duftler** [comment]
LGTM.

**Pull request merged**